File: problem-2-solution.js
function findMinimalSize(array) {
    let uniqueElements = new Set(array);
    return uniqueElements.size;
}

function main() {
    let testcase1 = [1, 2, 3, 2, 1];
    let testcase2 = [1, 1, 1, 1];
    
    console.log(findMinimalSize(testcase1)); // Output: 3
    console.log(findMinimalSize(testcase2)); // Output: 1
}

main();

File: problem-2.solution.java
import java.util.HashSet;

public class Main {
    public static int findMinimalSize(int[] array) {
        HashSet<Integer> uniqueElements = new HashSet<>();
        for (int num : array) {
            uniqueElements.add(num);
        }
        return uniqueElements.size();
    }

    public static void main(String[] args) {
        int[] testcase1 = {1, 2, 3, 2, 1};
        int[] testcase2 = {1, 1, 1, 1};

        System.out.println(findMinimalSize(testcase1)); // Output: 3
        System.out.println(findMinimalSize(testcase2)); // Output: 1
    }
}
